src/compiler/transformers/ts.ts

```diff
@@ -2657,18 +2654,22 @@ namespace ts {
         }
 
         /**
-         * Determines whether a declaration is the first declaration with the same name emitted
-         * in the current scope.
+         * Determines whether a declaration is *could* be the first declaration with
+         * the same name emitted in the current scope. Only returns false if we are absolutely
+         * certain a previous declaration has been emitted.
          */
-        function isFirstEmittedDeclarationInScope(node: Node) {
+        function isPotentiallyFirstEmittedDeclarationInScope(node: Node) {
+            // If the node has a named symbol, then we have enough knowledge to determine
+            // whether a prior declaration has been emitted.

```

Comment:
We should consider using `getParseTreeNode` here and get the symbol of the parse tree node. If there is no parse tree node, or the node does not have a symbol, we should fall back to the text of the declaration's name.